File: 백준/Silver/1072. 게임/게임.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').toString().trim().split('\n');
// const input = fs.readFileSync('input.txt').toString().trim().split('\n');
const [X, Y] = input[0].split(' ').map(Number);
const Z = Math.floor((Y * 100) / X);
let s = 1;
let e = 1000000000;
if (Z >= 99) {
  console.log(-1);
  process.exit(0);
}
while (s < e) {
  const mid = Math.floor((s + e) / 2);
  const rate = Math.floor(((Y + mid) * 100) / (X + mid));
  if (rate == Z) {
    s = mid + 1;
  } else {
    e = mid;
  }
}
console.log(s);

File: 백준/Silver/1072. 게임/게임.py
import sys
input = sys.stdin.readline

X, Y = map(int, input().split())
Z = int(Y*100/X)
if Z>=99:
    print(-1)
    exit()
s = 0
e = 1000000000

while s<e:
    mid = int((s+e)/2)
    rate = int((Y+mid)*100 / (X+mid))
    if rate == Z:
        s = mid+1
    else :
        e = mid
        
print(s)